internal/translation/dbuser/conversion.go

```diff
@@ -0,0 +1,221 @@
+package dbuser
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/timeutil"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type User struct {
+	*akov2.AtlasDatabaseUserSpec
+	Password  string
+	ProjectID string
+}
+
+// NewUser wraps a Kubernetes Atlas User Spec pointer augmenting it with projectID and password.
+func NewUser(spec *akov2.AtlasDatabaseUserSpec, projectID, password string) (*User, error) {
+	return normalize(&User{AtlasDatabaseUserSpec: spec, ProjectID: projectID, Password: password})
+}
+
+// DiffSpecs returns all differences found in the user Spec fields or a spec user and an atlas user.
+// Non Spec fields are not compared. Inputs are dbuser.User so they are normalized
+func DiffSpecs(specUser, atlasUser *User) []string {
```

Comment:
This looks very verbose and implies lots of boiler plate code for every type. Do we have to implement this for every type?

Given we only check for
```
	if len(diffs) > 0 {
		return true, nil
	}
```
what is the benefit vs. a `reflect.DeepEquals`?

Reply:
Precise debug info, this tells you exactly what was the first difference found and can save potentially a lot of debugging time later.
If we can achieve the same, or better like full diff reports, with less effort I am ll in.

Reply:
uhm , actually I was wrong, I forgot this is so precise that it lists ALL diffs found, something a mere boolean will not do.

Reply:
but why do we need the diff?

Reply:
my point is, in no place we evaluate the diff. we only check for the boolean result here: https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1543/files#diff-733b0371c07644c49998b08f8fbbdfd0827a48cf7c717754a972296f3b289f24R220

Reply:
My concern here is that this is a case of YAGNI. Precise diff was used in `cmp.Diff`, it is very inefficient and actually not used. If we want debug info, we can still add a debug log entry, serialized to json or yaml and print the diff.

The total cost of ownership in maintaining and mandating to maintain an exact diff is very high. If we avoid using `cmp.Diff` in the future this contribution mandates it to maintain it manually which is very verbose code.

Reply:
will make it a simple boolean equal and the debug could be after the fact and only on `false`